### DISCUSSION TITLE
Make it possible to choose a hero (og:image) in starlight without changing the layout

### DISCUSSION THREAD

**Discussion by tmcw** [withastro/starlight#1300] [Feature Requests]
### What version of `starlight` are you using?

0.15.1

### What is your idea?

I'd like to have `og:image` metadata on all pages that can have it, for infernal SEO reasons. Docs often don't have a good, splashy image, or benefit from having a splash layout. However, the `hero` frontmatter option controls _both_ og:image and the layout.

It would be nice for there to be something like

```yaml
hero:
  image: …
  hidden: true
```

### Why is this feature necessary?

For terrible SEO reasons,

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for your feedback.

> However, the `hero` frontmatter option controls _both_ og:image and the layout.

This is not the case. Starlight nevers defines an `og:image` meta tag, even when the `hero.image` is set.

To define an `og:image` meta tag, you don't have to change the layout or specify a `hero` image, and multiple options are available:

- Using the frontmater [`head` field](https://starlight.astro.build/reference/frontmatter/#head) to add a custom `<meta>` tag for `og:image` in each page and specify the image URL.
- Using the [global `head` configuration](https://starlight.astro.build/reference/configuration/#head) option to add a custom `<meta>` tag for `og:image` in all pages which would all use the same image URL.
- Use a [route data middleware](https://starlight.astro.build/guides/route-data/#customizing-route-data) to automatically compute the `og:image` URL based on the page's ID or path, and add it to the page's `head`.

For example, using a [route data middleware](https://starlight.astro.build/guides/route-data/#how-to-customize-route-data), you could do something like this:

```ts 
// src/routeData.ts
import { defineRouteMiddleware } from '@astrojs/starlight/route-data'

export const onRequest = defineRouteMiddleware((context) => {
  // Get the URL of the generated image for the current page using its ID and
  // append the `.png` file extension.
  const ogImageUrl = new URL(
    `/og/${context.locals.starlightRoute.id || 'index'}.png`,
    context.site,
  )

  // Get the array of all tags to include in the `<head>` of the current page.
  const { head } = context.locals.starlightRoute

  // Add the `<meta/>` tags for the Open Graph images.
  head.push({
    tag: 'meta',
    attrs: { property: 'og:image', content: ogImageUrl.href },
  })
  head.push({
    tag: 'meta',
    attrs: { name: 'twitter:image', content: ogImageUrl.href },
  })
})
```

With this approach, you could either generate such images or have them all in your `public/og/` directory, and they would be automatically included in the `<head>` of each page.
